Rework how the dependency coordinates found in templates are handled

melix opened the pull request:
Instead of using a pom.xml file, this now makes use of a Gradle version catalog, found in `gradle/templates.versions.toml`. This file should be handled by renovatebot automatically.

This will typically make the Gradle plugin upgrades much easier to handle since there will be a single place to change.

Fixes #1635

melix committed:
Rework how the dependency coordinates found in templates are handled

Instead of using a pom.xml file, this now makes use of a Gradle version
catalog, found in `gradle/templates.versions.toml`. This file should be
handled by renovatebot automatically.

This will typically make the Gradle plugin upgrades much easier to
handle since there will be a single place to change.

Fixes #1635

timyates reviewed (approved):
🧙

timyates commented:
I believe there's a fix in the pipeline from @sdelamo for the aws/groovy failures

melix commented:
This could be updated to Gradle plugin 3.7.4 but given the very slow feedback loop, I won't push this today.

melix committed:
Bump Gradle plugin version

sdelamo commented on starter-core/src/test/groovy/io/micronaut/starter/feature/dependencies/MultipleSourcesDependencyResolver.groovy:
```diff
@@ -4,16 +4,20 @@ import io.micronaut.context.annotation.Replaces
 @Replaces(PomDependencyVersionResolver.class)
 @Singleton
-class MultiplePomDependencyResolver implements PomDependencyVersionResolver {
+class MultipleSourcesDependencyResolver implements PomDependencyVersionResolver {
```
please, keep the old name to avoid an unnecessary breaking change.

melix replied:
I don't understand what breaking change you're talking about : this is a test.

sdelamo replied:
I did not realise was a test. Sorry for the noise.

sdelamo commented on starter-core/src/test/groovy/io/micronaut/starter/build/dependencies/PomCoordinatesUtils.java:
```diff
@@ -35,12 +35,12 @@
     private static final String NODE_NAME_TEXT = "#text";
 
-    private CoordinatesUtils() {
+    private PomCoordinatesUtils() {
```
please, keep the old name to avoid an unnecessary breaking change.

melix replied:
Same question here : this is a test, how can it be a breaking change?

melix replied:
Forget what I said, it's in `test` because that's where I moved the class.

sdelamo reviewed (changes requested):
Please, note that we also use the pom.xml in guides. https://github.com/micronaut-projects/micronaut-guides/blob/master/buildSrc/src/main/resources/pom.xml

I assume this is compatible to have another version catalogue in guides.

melix commented:
> we also use the pom.xml in guides. https://github.com/micronaut-projects/micronaut-guides/blob/master/buildSrc/src/main/resources/pom.xml

Oh, so you mean the `CoordinatesUtils` class is used in guides too? Then you're right I need to make changes to keep this working.

melix committed:
Restore `CoordinateUtils` for binary compatibility

This class is used in `micronaut-guides`, where something similar to
this PR could be done.
